web/src/Web.App/Services/FinanceService.cs

```diff
@@ -9,7 +12,34 @@ public interface IFinanceService
     Task<FinanceYears> GetYears();
 }
 
-public class FinanceService(IInsightApi insightApi) : IFinanceService
+public class FinanceService(
+    IInsightApi insightApi,
+    IMemoryCache memoryCache,
+    IOptions<CacheOptions> cacheOptions) : IFinanceService
```

Comment:
Would it be possible to type the options as `ReturnYearsCacheOptions` or similar, inheriting from `CacheOptions`? Then in the service configuration you can be explicit about which config section should be used to resolve the values in the case of multiple cache config sections in the future.

Reply:
Amended CacheOptions so it can be used as suggested